pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_PQSC.py

```diff
@@ -0,0 +1,242 @@
+                format(ZI_PQSC.MIN_FWREVISION, self.geti('system/fwrevision')))
+
+        if self.geti('system/fpgarevision') < ZI_PQSC.MIN_FPGAREVISION:
+            raise zibase.ziVersionError(
+                'Insufficient FPGA revision detected! Need {}, got {}!'.format(
+                    ZI_PQSC.MIN_FPGAREVISION,
+                    self.geti('system/fpgarevision')))
+
+    def _add_extra_parameters(self) -> None:
+        """
+        We add a few additional custom parameters on top of the ones defined in the device files. These are:
+          qas_0_trans_offset_weightfunction - an offset correction parameter for all weight functions,
+            this allows normalized calibration when performing cross-talk suppressed readout. The parameter
+            is not actually used in this driver, but in some of the support classes that make use of the driver.
+          AWG_file - allows the user to configure the AWG with a SeqC program from a specific file.
+            Provided only because the old version of the driver had this parameter. It is discouraged to use
+            it.
+          wait_dly - a parameter that enables the user to set a delay in AWG clocks cycles (4.44 ns) to be
+            applied between when the AWG starts playing the readout waveform, and when it triggers the
+            actual readout.
+          cases - a parameter that can be used to define which combination of readout waveforms to actually
+            download to the instrument. As the instrument has a limited amount of memory available, it is
+            not currently possible to store all 1024 possible combinations of readout waveforms that would
+            be required to address the maximum number of qubits supported by the instrument (10). Therefore,
+            the 'cases' mechanism is used to reduce that number to the combinations actually needed by
+            an experiment.
+        """
+        super()._add_extra_parameters()
+
+    # FIXME: put in correct clock_freq
+    def clock_freq(self):
+        return 300e6
+
+    ##########################################################################
+    # 'public' functions:
+    ##########################################################################
+
+    def check_errors(self, errors_to_ignore=None) -> None:
+        """
+        Checks the instrument for errors.
+        """
+        errors = json.loads(self.getv('raw/error/json/errors'))
+
+        # If this is the first time we are called, log the detected errors,
+        # but don't raise any exceptions
+        if self._errors is None:
+            raise_exceptions = False
+            self._errors = {}
+        else:
+            raise_exceptions = True
+
+        # Asserted in case errors were found
+        found_errors = False
+
+        # Combine errors_to_ignore with commandline
+        _errors_to_ignore = self._errors_to_ignore
+        if errors_to_ignore is not None:
+            _errors_to_ignore += errors_to_ignore
```

Comment:
Note that _errors_to_ignore is a reference to the mutable self._errors_to_ignore. This means that when calling
```
PQSC.check_errors(errors_to_ignore=['SOMEERROR'])
```
once, SOMEERROR will always be ignored afterwards. I guess that this is not the intended behavior and that something like
```
_errors_to_ignore = self._errors_to_ignore + errors_to_ignore if errors_to_ignore is not None else self._errors_to_ignore
```
should be used, right? (Or, if this is the intended behavior, it should be clarified in the docstring because it is quite unexpected from the user's perspective.)

The same issue exists in ZI_HDAWG_core.check_errors